kythe/rust/indexer/tests/providers.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2020 The Kythe Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate kythe_rust_indexer;
+use kythe_rust_indexer::{error::KytheError, providers::{FileProvider, KzipFileProvider}};
+use std::env;
+use std::fs::File;
+use std::path::{Path, PathBuf};
+
+/// Get the path of the provided runfile
+///
+/// # Panics
+///
+/// Will panic if the TEST_SRCDIR environment variable isn't set.
+fn get_runfile(file: &str) -> PathBuf {
+  let testfiles_path = "io_kythe/kythe/rust/indexer/testfiles/";
+  match env::var("TEST_SRCDIR") {
+    Ok(path) => {
+      let src_path = Path::new(&path);
```

Comment:
Rust novice here, looks like the Err(_) => panic! could be a call to .unwrap.

```
Path::new(env::var("TEST_SRCDIR").unwrap("TEST_SRCDIR envir..."))
    .join(testfiles_path)
    .join(file)
```

Reply:
Note that it's `expect` which takes an explanatory string (which is usually a good idea unless you're absolutely certain the condition is impossible).

Reply:
Good point. Thanks!